src/lib/rng/jitter_rng/jitter_rng.h

```diff
@@ -0,0 +1,40 @@
+/*
+* CPU Jitter Random Number Generator
+* (C) 2024 Planck Security S.A.
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_JITTER_RNG_H_
+#define BOTAN_JITTER_RNG_H_
+
+#include <botan/rng.h>
+
+namespace Botan {
+
+class BOTAN_DLL Jitter_RNG_Data;
```

Comment:
Considering that `Jitter_RNG_Data` is only used internally and offers no API available to the user whatsoever, I believe its definition would be better placed inside `Jitter_RNG`. This way there will be one less element polluting the `Botan` namespace.

Reply:
Makes perfect sense, updated.

Reply:
I like your train of thought in general (of not polluting the API further). However, if we declare it as a free-standing forward declare (without `BOTAN_DLL`), we can hide the implementation from public linkage within the cpp-file compilation unit. This is fairly common practice throughout the code base.

See also: https://github.com/randombit/botan/pull/4325#discussion_r1784634554

Reply:
continuation: https://github.com/randombit/botan/pull/4325#discussion_r1789859848